### PR TITLE
feat(web): recent searches history — one-click rerun of prior lookups

### LINKED ISSUE #264: web: recent searches history — one-click rerun of prior lookups
## Problem

Iterating on a card list ("did I forget Mew? what about Mewtwo?") means retyping or re-pasting the same lines over and over. The current SPA has zero memory of past runs.

## Proposal

A collapsible **Recent** panel under the input editor showing the last N (default 10) bulk runs:

- Each entry: timestamp, line count, top-of-list preview ("Charizard, Pikachu, +8 more"), the same chip rendering used in the empty-state examples.
- One-click rerun puts the lines back in the editor and triggers a lookup.
- Hover/long-press → delete. Clear-all in the panel header.
- Stored in Zustand with persistence (capped at N to keep localStorage small).

## Acceptance

- [ ] Panel renders below the input editor with the last N runs.
- [ ] Clicking an entry restores the lines and runs the lookup.
- [ ] Delete + clear-all work, persistence respects them.
- [ ] Tests cover capping at N, restore-and-rerun, delete.

### PR DESCRIPTION
**Pull request by mgzwarrior**
Closes #264.

## What

Adds a collapsible **Recent searches** panel under the input editor
showing the last 10 bulk-lookup submissions. Each entry carries:

- A relative timestamp (`just now`, `5m ago`, `2h ago`, `3d ago`)
- A line-count badge (`4 lines` / `1 line` with correct pluralization)
- A `Charizard, Pikachu, +3 more` preview built from the first two lines

One click on a row restores the lines into the editor and reruns the
lookup. Hover a row for a `×` to delete it; **Clear all** in the
panel header wipes the whole list.

Backed by a new Zustand slice (`recentRuns`, `pushRecentRun`,
`removeRecentRun`, `clearRecentRuns`) capped at `RECENT_RUNS_LIMIT`
(10) and persisted via the existing `partialize` middleware so the
history survives a page reload. Consecutive duplicate submissions
collapse by refreshing the head entry's `savedAt` rather than
stacking copies — re-running the same list twice in a row doesn't
clutter the panel.

## Why

Iterating on a card list — "did I forget Mew? what about Mewtwo?" —
meant retyping or re-pasting the same lines over and over. The SPA
had zero memory of past runs. The Recent panel sits adjacent to the
input editor and turns "I just had that list" into a one-click
rerun.

`pushRecentRun` is wired into `handleRun` at click time (not on
completion), so a run the user stops or that errors still leaves a
re-runnable entry behind — exactly when you want to come back to it.

## How to verify

1. `make dev-api` + `make dev-web`.
2. Submit a few different card lists.
3. Confirm:
   - A **Recent searches (N)** panel renders below the textarea.
   - Each entry shows a relative timestamp, line count, and preview.
   - Clicking an entry restores the lines and reruns the lookup.
   - Hover an entry → the `×` appears, click deletes just that entry.
   - **Clear all** in the header wipes the list.
   - Collapse arrow hides entries while keeping the header counter.
   - Reload the page — the history is still there.
   - Submit the same list twice — only one entry is recorded.

## Proof

Verified locally via the preview server:

- 2 distinct submissions → 2 panel entries, newest at top
- Click rerun on the older Pikachu entry → lines restored, lookup ran
- Delete on the 4-line entry → removed cleanly
- Clear all → list emptied
- Reload after a fresh submission → entry persisted

Screenshot from the running preview attached after open. Happy to
add a Jam clip if reviewers prefer the motion.

## Tests

- `store/index.test.ts`: prepends new entries with `id`/`savedAt`;
  ignores empty submissions; collapses consecutive duplicates by
  refreshing `savedAt`; caps at `RECENT_RUNS_LIMIT`; `removeRecentRun`
  drops by id; `clearRecentRuns` empties.
- `components/RecentRuns.test.tsx`: renders nothing on empty list;
  per-entry line-count + preview rendering; click restores +
  triggers `onRun`; no-op while a run is already in flight; per-row
  delete; clear-all; collapse hides entries but keeps the counter.

All 141 web tests pass; `make check` + `npm run build` green.

### CODE REVIEW

File: web/src/utils/format.ts [hunk 1]
```diff
@@ -32,6 +32,28 @@ export function formatComp(
...
+ * suitable for the recent-searches panel: `just now`, `5m ago`,
+ * `2h ago`, `3d ago`, or an absolute `MMM D` for older entries. The
+ * `now` argument is injectable so tests can pin the boundary cases.
+ */
+export function formatRelativeTime(timestamp: number, now: number = Date.now()): string {
+  const delta = Math.max(0, now - timestamp)
+  const seconds = delta / 1000
+  if (seconds < 45) return 'just now'
+  const minutes = seconds / 60
+  if (minutes < 60) return `${Math.round(minutes)}m ago`
```
[the remark this answers is not in the document]
Comment (mgzwarrior) [reply web/src/utils/format.ts]: Switched to `Math.max(1, Math.floor(...))` for the minute/hour/day buckets in 5465a3a so 59.6 min → `59m ago` instead of `60m ago`.

File: web/src/utils/format.ts [hunk 2]
```diff
@@ -32,6 +32,28 @@ export function formatComp(
  * round to exactly `60.00s` (e.g. 59_995 ms) roll over to `1m00s` so
  * the sub-minute branch never displays the next minute's boundary.
  */
+/**
+ * Format a wall-clock timestamp as a compact relative-time string
+ * suitable for the recent-searches panel: `just now`, `5m ago`,
+ * `2h ago`, `3d ago`, or an absolute `MMM D` for older entries. The
+ * `now` argument is injectable so tests can pin the boundary cases.
```
[the remark this answers is not in the document]
Comment (mgzwarrior) [reply web/src/utils/format.ts:31]: Moved the `formatElapsed` JSDoc back onto `formatElapsed` in 5465a3a — it had drifted to attach to `formatRelativeTime`.

File: web/src/components/RecentRuns.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          {run.lines.length} {lineWord}
+        </span>
+        <span className="truncate font-mono text-zinc-300">{summary}</span>
+      </button>
+      <button
+        type="button"
+        onClick={onDelete}
+        aria-label={`Delete recent search: ${summary}`}
+        className="flex-shrink-0 rounded p-1 text-zinc-500 opacity-0 group-hover:opacity-100 hover:text-zinc-200 hover:bg-zinc-700 transition-opacity"
+      >
```
[the remark this answers is not in the document]
Comment (mgzwarrior) [reply web/src/components/RecentRuns.tsx:120]: Made the delete `×` visible by default on touch viewports and via `focus-visible` / `group-focus-within` in 5465a3a; hidden-until-hover is now `sm:` and up only.
